_posts/2017/2017-04-06-javaland-2017-in-phantasialand.md

```diff
@@ -0,0 +1,75 @@
+He extended the observer pattern and added push-based streams, which can be used when it is unknown when data is available.
+At its core the [Spring Reactor](fully non-blocking foundation with efficient demand management), a fully non-blocking foundation with efficient demand management, is used.
+Another important role plays [Flux](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html) which is similar to the [Java 8 Stream API](https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html).
+A new paradigm in Spring Land will be released in June and is called [Spring Web Flux](https://spring.io/blog/2017/03/15/spring-tips-the-spring-web-flux-reactive-client).
+See [this video](https://www.youtube.com/watch?v=leZdgr-O4LE) by [Josh Long](https://twitter.com/starbuxman) for more details.
+
+## Migrating to Java 9
+
+In [this talk](https://programm.javaland.eu/2017/talk.html#talk?talkId=529430) by [Sander Mak](https://twitter.com/sander_mak), who is currently writing a book with [Paul Bakker](https://twitter.com/pbakker) about [Java 9 Modularity](https://twitter.com/javamodularity), explained in this talk the essential steps to prepare for the next Java major release.
+Use [jdeps](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/jdeps.html) to get information about missing platform modules.
+This tool has some interesting option to inspect JDK internals like and some you should not use like `—add-exports` and `—permit-illegal-access `.
+Better change your could instead of using these flags.
+If you want to migrate to use modules you can put a plain jair on the module path (automatic modules).
+The name is derived from the jar name and export everthing needed as well as reads other downstream modules.
+It can access the class path. You should create a module which requires `commons.lang` and a `module-info.jar`.
+Dependencies have to be module themselves for easier handling (transitive dependencies).
+At runtime services (components) are accessible via reflection.
+Don’t open module to everyone.
+Export only certain api packages including services!
+Revisit open modules when you are done.
+Create several modules that are logically seperated, e.g. `API`, `Impl` and `Main`.
+
+# Workshop Day
+
+At the [workshop day](https://www.javaland.eu/de/programm/schulungstag) we visited the Java Web Security session by [Dominik Shadow](https://twitter.com/dschadow) ([Website](https://blog.dominikschadow.de/)) – material can be found [here](https://github.com/dschadow/JavaSecurity) – and the Hibernate 4 session by [Thorben Jansen](https://twitter.com/thjanssen123), who just released a new [Hibernate Tips book](http://www.thoughts-on-java.org/hibernate-tips-book-release-special-launch-price/) about it and run a very popular web portal about Java and Hibernate - [www.thoughts-on-java.org](http://www.thoughts-on-java.org/).
```

Comment:
"At the [workshop day](https://www.javaland.eu/de/programm/schulungstag) we visited the [Java Web Security](https://github.com/dschadow/JavaSecurity) by [Dominik Shadow](https://twitter.com/dschadow) ([Website](https://blog.dominikschadow.de/)) and the Hibernate 4 session by...
...and who runs a very popular [web portal](http://www.thoughts-on-java.org/) about Java and Hibernate."

Is there also a link to the Hibernate 4 session?